### PR TITLE
feat: Capture latency and token usage for agentic run

### PR DESCRIPTION
**Pull request by asamal4**
## Description

feat: Capture latency and token usage for agentic run
Each run will sum all the stages for token, for latency max timestamp - min timestamp

## Type of change

- [ ] Refactor
- [x] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up service version
- [ ] Bump-up dependent library
- [ ] Bump-up library or tool used for development (does not change the final image)
- [ ] CI configuration change
- [ ] Unit tests improvement

## Tools used to create PR

Identify any AI code assistants used in this PR (for transparency and review context)

- Assisted-by: (e.g., Claude, CodeRabbit, Ollama, etc., N/A if not used)
- Generated by: (e.g., tool name and version; N/A if not used)

## Related Tickets & Documents

- Related Issue #
- Closes #

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] PR has passed all pre-merge test jobs.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- Please provide detailed steps to perform tests related to this code change.
- How were the fix/results from this change verified? Please provide relevant screenshots or results.

## Summary by CodeRabbit

* **New Features**
  * Added token usage tracking across evaluation stages, including aggregated input and output totals.
  * Added agent latency reporting based on the elapsed time between stage transitions.
  * Supports usage and timing data from all evaluation stages, including non-analysis stages.

* **Bug Fixes**
  * Missing or invalid usage and timing data now falls back safely without interrupting evaluation results.

### CODE REVIEW

File: src/lightspeed_evaluation/pipeline/evaluation/openshift_agentic_run_amender.py [hunk 1]
```diff
@@ -81,10 +82,61 @@ def _do_amend(
...
+                # Collect timestamps
+                conditions = result.get("conditions", [])
+                for cond in conditions:
+                    if not isinstance(cond, dict):
+                        continue
+                    ts_str = cond.get("lastTransitionTime")
+                    if not ts_str:
+                        continue
+                    try:
+                        all_timestamps.append(datetime.fromisoformat(ts_str))
```
[the remark this answers is not in the document]
Comment (asamal4) [reply src/lightspeed_evaluation/pipeline/evaluation/openshift_agentic_run_amender.py:125]: is it possible with same implementation/application but different timestamp ?

File: src/lightspeed_evaluation/pipeline/evaluation/openshift_agentic_run_amender.py [hunk 2]
```diff
@@ -81,10 +82,61 @@ def _do_amend(
...
+        all_timestamps: list[datetime] = []
+
+        for step_name in STEP_RESOURCES:
+            step_results = results.get(step_name, [])
+            for result in step_results:
+                # Collect token usage
+                token_usage = result.get("tokenUsage", {})
+                if token_usage:
+                    total_input_tokens += token_usage.get("inputTokens", 0)
+                    total_output_tokens += token_usage.get("outputTokens", 0)
```
[the remark this answers is not in the document]
Comment (asamal4) [reply src/lightspeed_evaluation/pipeline/evaluation/openshift_agentic_run_amender.py:114]: is this edge case possible ? - this data is not added by user, rather by application

File: src/lightspeed_evaluation/pipeline/evaluation/openshift_agentic_run_amender.py [hunk 3]
```diff
@@ -81,10 +82,61 @@ def _do_amend(
         turn_data.openshift_agentic_run_phases = [
             step for step in STEP_RESOURCES if results.get(step)
         ]
+        self._populate_token_usage_and_latency(turn_data, results)
```
[the remark this answers is not in the document]
Comment (asamal4) [reply src/lightspeed_evaluation/pipeline/evaluation/openshift_agentic_run_amender.py:85]: how it will use previous data ? each turn gets it's own object with token/latency initialized to 0

File: tests/unit/pipeline/evaluation/test_openshift_agentic_run_amender.py [hunk 4]
```diff
@@ -72,9 +72,20 @@ def _get_response(turn: TurnData) -> str:
...
+            "type": "Started",
+            "status": "True",
+            "reason": "StepStarted",
+            "lastTransitionTime": "2026-09-18T10:00:00Z",
+        },
+        {
+            "type": "Completed",
+            "status": "True",
+            "reason": "Succeeded",
+            "lastTransitionTime": "2026-09-18T10:00:20Z",
```
[the remark this answers is not in the document]
Comment (asamal4) [reply tests/unit/pipeline/evaluation/test_openshift_agentic_run_amender.py]: fixing the test data..

File: tests/unit/pipeline/evaluation/test_openshift_agentic_run_amender.py [hunk 5]
```diff
@@ -608,6 +641,230 @@ def test_subprocess_error_caught_by_amend(self, mocker: MockerFixture) -> None:
...
+                "execution": {"results": [{"name": "er-1", "outcome": "Succeeded"}]},
+                "verification": {"results": [{"name": "vr-1", "outcome": "Succeeded"}]},
+            },
+        }
+
+        amender.amend(turn, status)
+
+        # From 2026-09-01T13:27:01.248893Z to 2026-09-01T13:32:35.599737Z
+        # (earliest analysis start to latest verification completion) ≈ 334.35 seconds
+        assert turn.agent_latency == pytest.approx(334.35, abs=0.1)
```
[the remark this answers is not in the document]
Comment (asamal4) [reply tests/unit/pipeline/evaluation/test_openshift_agentic_run_amender.py:736]: this is test data, also stages are not manadatory

File: tests/unit/pipeline/evaluation/test_openshift_agentic_run_amender.py [hunk 6]
```diff
@@ -608,6 +641,230 @@ def test_subprocess_error_caught_by_amend(self, mocker: MockerFixture) -> None:
...
+
+    def test_token_usage_zero_when_missing(self) -> None:
+        """Test token usage defaults to 0 when tokenUsage field is absent."""
+        status_no_tokens: dict[str, Any] = {
```
[the remark this answers is not in the document]
Comment (asamal4) [reply tests/unit/pipeline/evaluation/test_openshift_agentic_run_amender.py:740]: field is populated by application, not user.. unnecessary edge cases
